home/whatsapp.py

```diff
@@ -99,37 +99,131 @@ def from_locale(cls, locale: Locale) -> "WhatsAppLanguage":
         return cls[lc]
 
 
-def create_whatsapp_template(
+def get_upload_session_id(image_obj: Image) -> dict[str, Any]:
+    """
+    Gets a session ID from the Turn API, to use with an image upload
+    """
+    url = urljoin(
+        settings.WHATSAPP_API_URL,
+        "graph/v14.0/app/uploads",
+    )
+    headers = {
+        "Content-Type": "application/json",
+    }
+    mime_type = mimetypes.guess_type(image_obj.file.name)[0]
+    file_size = image_obj.file.size
+
+    data = {
+        "file_length": file_size,
+        "file_type": mime_type,
+        "access_token": settings.WHATSAPP_ACCESS_TOKEN,
+        "number": settings.FB_BUSINESS_ID,
+    }
+    response = requests.post(
+        url,
+        headers=headers,
+        data=json.dumps(data, indent=4),
+    )
+
+    upload_details = {
+        "upload_session_id": response.json()["id"],
+        "upload_file": image_obj.file,
+    }
+
+    response.raise_for_status()
```

Comment:
We probably want to call `raise_for_status` before we call `.json()`, otherwise we're trying to decode the response before we start checking if it was a success.

Reply:
👍